libnmstate/schemas/operational-state.yaml

```diff
@@ -408,3 +416,49 @@ definitions:
                     type: boolean
                   state:
                     type: string
+  rib:
+    type: object
+    properties:
+      name:
+        type: string
+      route-table-id:
+        type: integer
+      address-family:
+        type: string
+        enum:
+          - ipv4
+          - ipv6
+      default-rib:
+        type: boolean
+      routes:
+        type: array
+        items:
+          type: object
+          $ref: "#/definitions/route"
+  route:
+    type: object
+    properties:
+      source-protocol:
+        type: string
+        enum:
+          - direct
+          - static
+          - dhcp        # DHCP RFC 3442
+          - ipv6-ra     # IPv6 Router Advertisement
+      destination-prefix:
+        type: string
+      next-hop:
+        type: object
+        $ref: "#/definitions/route-next-hop"
+      route-preference:
+        type: integer
+  route-next-hop:
```

Comment:
I think it would be simpler to use the "union" used in ipv4: prefix-length and netmask.
The use can specify one of them.

Reply:
I don't get it, can you elaborate?